integration-tests/tests/src/tests/sync/user.ts

```diff
@@ -0,0 +1,620 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+import { expect } from "chai";
+import { importAppBefore } from "../../hooks";
+import {
+  randomNonVerifiableEmail,
+  randomPendingVerificationEmail,
+  randomVerifiableEmail,
+  uuid,
+} from "../../utils/generators";
+import { KJUR } from "jsrsasign";
+import { sleep } from "../../utils/sleep";
+
+function assertIsUser(user: Realm.User) {
```

Comment:
A very small thing: Technically it does an assertion, but you might want to lend the wording from `chai` here.

```suggestion
function expectIsUser(user: Realm.User) {
```